test/fuzz/oss-fuzz-build.sh

```diff
@@ -2,18 +2,27 @@
 
 set -euo pipefail
 
+# Upgrade to Go 1.18. Remove when it's the default.
```

Comment:
I'm not sure where this is intended to run. If it's meant to run locally, then we should not presume the existence of the `apt-get` toolchain, since that will not exist on Darwin or many container bases (e.g., Alpine).

I don't see it being invoked from CI—maybe I missed something? But in any case, if we want to insist on installing something, we should probably use ambient tools and report an error if something is missing.

Reply:
This is a build script specialized for OSS-Fuzz, called here: https://github.com/google/oss-fuzz/blob/d533911bf33c3d90f23b5d3363d6d89a3a0f0e54/projects/tendermint/build.sh#L18 . The suggestion is from https://github.com/AdamKorcz/go-118-fuzz-build/commit/7bed8f92ea820f668efab099ec774d548f1873a8#commitcomment-75092178 .

OSS-Fuzz is in the process of switching to Go 1.18 by default and working on adding native support for Go 1.18 fuzzing. That's why we have to do these shenanigans.

Reply:
That makes sense, thanks for the explanation. Can we add a comment to this file to record that relationship?